_posts/2015-04-24-FFI-and-Rust.md

```diff
@@ -0,0 +1,264 @@
+---
+layout: post
+title: "FFI and Rust"
+author: Alex Crichton
+description: "Zero-cost and safe FFI in Rust"
+---
+
+
+Rust's quest for world domination was never destined to happen overnight, so
+Rust needs to be able to interoperate with the the existing world just as easily
+as it talks to itself. To solve this problem, **Rust lets you communicate with C
+APIs at no extra cost while providing strong safety guarantees**.
+
+This is also referred to as Rust's foreign function interface (FFI) and is the
+method by which Rust communicates with other programming languages. In following
+Rust's design principles, this is a **zero cost abstraction** which leverages
+language features such as ownership and borrowing to provide a **safe
+interface**.
+
+In this post we'll explore how Rust's zero-cost FFI and ability to write safe
+abstractions by looking at some examples of interacting with C. C is, however,
+just an example, as we'll also see how Rust can easily talk to languages like
+Python and Ruby just as seamlessly as C.
+
+### Talking to C
+
+First, let's start with an example of calling C code from Rust and then take a
+look at how there is no overhead. Starting off simple, here's a C program which
```

Comment:
Instead of 'take a look at how' you might say 'demonstrate that'. Instead of just 'no overhead', which is somewhat ambiguous, maybe 'Rust imposes no additional performance overhead', or 'the machine code generated by Rust to call C is equivalent to that generated by a C compiler'.